### PR TITLE
[1] Add test for validation results

### PR DESCRIPTION
**Pull request by dczulada**
Pull requests into Cypress require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [ ] This pull request describes why these changes were made.
- [ ] Internal ticket for this PR:
- [ ] Internal ticket links to this PR
- [ ] Code diff has been done and been reviewed
- [ ] Tests are included and test edge cases
- [ ] Tests have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

**Comment by dczulada** [comment]
Here is a link to CodeCov
https://app.codecov.io/gh/projecttacoma/cvu-fhir/compare/12/overview
Now 91.09%

File: test/controllers/vendor_records_controller_test.rb [hunk 1]
```diff
@@ -98,6 +98,29 @@ class VendorsRecordsControllerTest < ActionController::TestCase
     end
   end
 
+  test 'should validate vendor patients' do
+    for_each_logged_in_user([ADMIN]) do
+      # Clear out old vendor patients (loaded from factories)
+      @vendor.fhir_patient_bundles.destroy
+
+      filename = Rails.root.join('test', 'fixtures', 'artifacts', 'good_patient_upload.zip')
```
Comment (lmd59) [inline test/controllers/vendor_records_controller_test.rb]: minor comment that 'good_patient_upload.zip' is a confusing name for a zip with patients with validation errors.

File: app/helpers/patient_analysis_helper.rb [hunk 2]
```diff
@@ -1,238 +1,5 @@
...
-    # find the number of code systems, codes, and percent code coverage per valueset for all valuesets
-    total_vs_code_sys = {}
-    total_vs_codes = {}
-    percent_vs_codes = {}
-    per_vs_stats(total_vs, total_vs_code_sys, total_vs_codes, vs_codes_found, percent_vs_codes)
-    # create analysis hash
-    analysis.merge!(advanced_valueset_analysis(total_vs, value_sets_found, total_vs_code_sys, vs_code_sys_found, percent_vs_codes))
-    analysis.merge!(coverage_analysis(measures_found))
-    analysis
+  def generate_analysis(_patients, _measure, _bundle)
```
Comment (lmd59) [inline app/helpers/patient_analysis_helper.rb:2]: do we want to stick a TODO here?

**Review by lmd59 — APPROVED** [review]
